File: TireShopFrontend/src/types/Image.ts
export interface Image {
  id: number;
  url: string;
  productId: number;
  publicId: string;
}

export interface CreateImageRequest {
  url: string;
  productId: number;
  publicId: string;
}

export interface AddImageRequest {
  url: string;
  productId: number;
  publicId: string;
}

File: TiresShopApp/src/main/java/org/tireshop/tiresshopapp/entity/Image.java
package org.tireshop.tiresshopapp.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "image")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Image {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  private String url;

  @ManyToOne
  @JoinColumn(name = "product_id")
  private Product product;

  private String publicId;
}
